weaviate_agents/query/query_agent.py

```diff
@@ -186,13 +216,87 @@ def stream(
             Union[ProgressMessage, StreamedTokens, QueryAgentResponse], None
         ],
     ]:
-        """Stream from the query agent. Must be implemented by subclasses."""
+        """Stream from the query agent.
+
+        Deprecated:
+            The `stream` method is deprecated; use `ask_stream()` instead.
+        """
+        pass
+
+    @overload
+    def ask_stream(
+        self,
+        query: Union[str, list[ChatMessage]],
+        collections: Union[list[Union[str, QueryAgentCollectionConfig]], None] = None,
+        include_progress: Literal[True] = True,
+        include_final_state: Literal[True] = True,
+    ) -> Union[
+        Generator[
+            Union[ProgressMessage, StreamedTokens, QueryAgentResponse], None, None
+        ],
+        AsyncGenerator[
+            Union[ProgressMessage, StreamedTokens, QueryAgentResponse], None
+        ],
+    ]: ...
+
+    @overload
+    def ask_stream(
+        self,
+        query: Union[str, list[ChatMessage]],
+        collections: Union[list[Union[str, QueryAgentCollectionConfig]], None] = None,
+        include_progress: Literal[True] = True,
+        include_final_state: Literal[False] = False,
+    ) -> Union[
+        Generator[Union[ProgressMessage, StreamedTokens], None, None],
+        AsyncGenerator[Union[ProgressMessage, StreamedTokens], None],
+    ]: ...
+
+    @overload
+    def ask_stream(
+        self,
+        query: Union[str, list[ChatMessage]],
+        collections: Union[list[Union[str, QueryAgentCollectionConfig]], None] = None,
+        include_progress: Literal[False] = False,
+        include_final_state: Literal[True] = True,
+    ) -> Union[
+        Generator[Union[StreamedTokens, QueryAgentResponse], None, None],
+        AsyncGenerator[Union[StreamedTokens, QueryAgentResponse], None],
+    ]: ...
+
+    @overload
+    def ask_stream(
+        self,
+        query: Union[str, list[ChatMessage]],
+        collections: Union[list[Union[str, QueryAgentCollectionConfig]], None] = None,
+        include_progress: Literal[False] = False,
+        include_final_state: Literal[False] = False,
+    ) -> Union[
+        Generator[StreamedTokens, None, None],
+        AsyncGenerator[StreamedTokens, None],
+    ]: ...
+
+    @abstractmethod
+    def ask_stream(
+        self,
+        query: Union[str, list[ChatMessage]],
+        collections: Union[list[Union[str, QueryAgentCollectionConfig]], None] = None,
+        include_progress: bool = True,
+        include_final_state: bool = True,
+    ) -> Union[
+        Generator[
+            Union[ProgressMessage, StreamedTokens, QueryAgentResponse], None, None
+        ],
+        AsyncGenerator[
+            Union[ProgressMessage, StreamedTokens, QueryAgentResponse], None
+        ],
+    ]:
+        """Run the Query Agent ask mode and stream the response."""
         pass
 
     @abstractmethod
     def search(
         self,
-        query: str,
+        query: Union[str, list[ChatMessage]],
```

Comment:
Does it make sense for `search` mode to receive conversational context the same as `ask` mode?